Updated changelogs for version 1.0.2 (#379)
awslabs/aws-advanced-jdbc-wrapper: CHANGELOG.md

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/), and this project adheres to [Semantic Versioning](https://semver.org/#semantic-versioning-200).

## [1.0.2] - 2023-03-30
### :magic_wand: Added
- Read-write splitting plugin example using Spring JDBC template. See example [here](./examples/AWSDriverExample/src/main/java/software/amazon/ReadWriteSplittingSpringJdbcTemplateExample.java)
- Read-write splitting connection plugin benchmark and performance results ([PR #340](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/340) and [PR #316](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/316))

> - `auroraConnectionTracker` closes opened connections pointing to a dead node ([PR #298](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/298))

Reviewer: ```suggestion
- `auroraConnectionTracker` tracks all opened connections and closes all impacted connections after a failover ([PR #298](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/298))
```

Reviewer: `auroraConnectionTracker` is the wrapperPlugin code, let's use the full plugin name instead to stay consistent with how we refer to `DriverMetaDataConnectionPlugin`

- `DriverMetaDataConnectionPlugin` allows users to override the driver name ([PR #371](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/371))
- Default list of plugins added to the parameter `wrapperPlugins` ([PR #332](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/332))
- Documentation for `DriverMetadataConnectionPlugin` and `auroraConnectionTracker` ([PR #378](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/378) and ([PR #360](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/360)))

### :bug: Fixed
- Unwrap Savepoint objects when passing them in as parameters ([Issue #328](https://github.com/awslabs/aws-advanced-jdbc-wrapper/issues/328))

> - Add null checks to WrapperUtils.getConnectionFromSqlObject() ([Issue #348](https://github.com/awslabs/aws-advanced-jdbc-wrapper/issues/348))

Reviewer: Lets be consistent with the tenses.

### :crab: Changed

> - Locked initialization of `AuroraHostListProvider` ([PR #347](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/347))

Reviewer: ```suggestion
- Lock initialization of `AuroraHostListProvider` ([PR #347](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/347))
```

> -  Optimized thread locks and expiring cache for the Enhanced Monitoring Plugin. ([PR #365](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/365))

Reviewer: ```suggestion
- Optimized thread locks and expiring cache for the Enhanced Monitoring Plugin. ([PR #365](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/365))
```

- Hibernate integration test ([PR #352](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/352))
- Documentation for the new connection plugin DriverMetadataConnectionPlugin and auroraConnectionTracker ([PR #378](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/378) and ([PR #360](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/360/files)))

### :bug: Fixed
- Close opened connections pointing to a dead node ([PR #298](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/298))
- Unwrap Savepoint objects when passing them in as parameters ([PR #331](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/331))
- Add null checks to WrapperUtils.getConnectionFromSqlObject() ([PR #348](https://github.com/awslabs/aws-advanced-jdbc-wrapper/issues/348))

### :crab: Changed
- Lock initialization of AuroraHostListProvider ([PR #347](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/347))
- Optimize thread locks and cache for EFM plugin ([PR #365](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/365))

> - Modify setReadOnly(true) failure scenarios ([PR #356](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/356))

Reviewer 2: ```suggestion
- Modified `setReadOnly(true)` failure scenarios ([PR #356](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/356))
```

> - Update Hibernate sample code to reflect changes in the wrapper source code ([PR #368](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/368))

Reviewer 2: ```suggestion
- Updated Hibernate sample code to reflect changes in the wrapper source code ([PR #368](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/368))
```

> - Docs now specify the wrapper driver does not support blue/green deployments ([PR #376](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/376))

Reviewer 2: I would suggest we use the same changelog sentence from the AWS JDBC Driver for MySQL. In addition, could we link to the README.md section containing this change, instead of the PR? I think we should do this for all documentation changes as that anchors customers to that specific section.

```suggestion
- Updated README.md to reflect that Amazon RDS Blue/Green Deployments are not supported.
```

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/), and this project adheres to [Semantic Versioning](https://semver.org/#semantic-versioning-200).

## [1.0.2] - 2023-01-30
### :magic_wand: Added
- New plugin allowing users to override the driver name ([PR #371](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/371))
- Default list of plugins added to the parameter 'wrapperPlugins' ([PR #332](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/332))

> - 2-node test cluster configuration ([PR #345](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/345))

Reviewer 2: To @karenc-bq 's point, we can omit test changes here.

> - Read-write splitting plugin example using Spring JDBC template. See example [here](./examples/AWSDriverExample/src/main/java/software/amazon/ReadWriteSplittingSpringJdbcTemplateExample.java)

Reviewer: Can we visually group similar changes? E.g. put the r/w splitting changes near each other.

> - Benchmark tests for read write splitting plugin ([PR #317](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/317))

Reviewer: We can keep this point about benchmarks, but could you please link the benchmark results or group them with the point about r/w perf results.

Reviewer: Nevermind I see a point about the results, you can leave this point out.

- Tests to ensure connection properties from the JDBC URL don't override existing credentials when reusing the AWSWrapperDataSource object ([PR #337](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/337))
- Test if AWS supported region url exists as reported in issue [Issue #344](https://github.com/awslabs/aws-advanced-jdbc-wrapper/issues/344). See test in [PR #351](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/351)
- Hibernate integration test ([PR #352](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/352))

> - Remaining tests added to refactored test suite and removed old test suite ([PR #364](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/364)) and ([PR #377](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/377))

Reviewer: We can leave testing changes out.

> - Read-write splitting and reader load balancing plugin results ([PR #340](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/340))

Reviewer: Read-write splitting connection plugin performance results

- Read-write splitting plugin benchmark and performance test results added to README ([PR #316](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/316))
- Documentation for the new connection plugin DriverMetadataConnectionPlugin and auroraConnectionTracker ([PR #378](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/378) and ([PR #360](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/360/files)))

### :bug: Fixed
- Close opened connections pointing to a dead node ([PR #298](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/298))
- Unwrap Savepoint objects when passing them in as parameters ([PR #331](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/331))

> - Add null checks to WrapperUtils.getConnectionFromSqlObject() ([PR #348](https://github.com/awslabs/aws-advanced-jdbc-wrapper/issues/348))

Reviewer: The first point actually adds a new connection plugin. I suggest we reframe this and put it to the "added" list.

### :crab: Changed
- Lock initialization of AuroraHostListProvider ([PR #347](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/347))
- Optimize thread locks and cache for EFM plugin ([PR #365](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/365))
- Modify setReadOnly(true) failure scenarios ([PR #356](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/356))
- Update Hibernate sample code to reflect changes in the wrapper source code ([PR #368](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/368))

> - Docs now specify the wrapper driver does not support blue/green deployments ([PR #376](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/376))

Reviewer: Lets avoid abbreviation.
